frontend/src/components/common/Header.tsx

```diff
@@ -8,51 +9,163 @@ import {
   Flex,
   Icon,
   IconButton,
+  Image,
   Link,
   Stack,
   Text,
   useDisclosure,
 } from "@chakra-ui/react";
-import React from "react";
-import { Link as ReactLink } from "react-router-dom";
+import React, { useContext, useEffect } from "react";
+import { Link as ReactLink, useHistory } from "react-router-dom";
 
-import { LOGIN_PAGE } from "../../constants/Routes";
-import { MenuIcon } from "./icons";
+import authAPIClient from "../../APIClients/AuthAPIClient";
+import {
+  HOME_PAGE,
+  LANDING_PAGE,
+  LOGIN_PAGE,
+  SCHEDULING_PAGE,
+} from "../../constants/Routes";
+import AuthContext from "../../contexts/AuthContext";
+import { CloseIcon, MenuIcon } from "./icons";
 
 const Header = (): JSX.Element => {
+  const history = useHistory();
   const { isOpen, onOpen, onClose } = useDisclosure();
+  const { authenticatedUser, setAuthenticatedUser } = useContext(AuthContext);
+
+  const onLogOutClick = async () => {
+    const success = await authAPIClient.logout(authenticatedUser?.id);
+    if (success) {
+      setAuthenticatedUser(null);
+      history.push(LANDING_PAGE);
```

Comment:
why do we need this if we're already on `/`?
```suggestion
```

Reply:
ahh I thought we could redirect to the landing page on successful sign out, I think we can leave this out for now though